helix-core/src/test/java/org/apache/helix/controller/rebalancer/waged/TestWagedRebalancerMetrics.java

```diff
@@ -85,7 +91,57 @@ public void testMetricValuePropagation()
 
     // Check that there exists a non-zero value in the metrics
     Assert.assertTrue(_metricCollector.getMetricMap().values().stream()
-        .anyMatch(metric -> metric.getLastEmittedMetricValue() > 0L));
+        .anyMatch(metric -> (long) metric.getLastEmittedMetricValue() > 0L));
+  }
+
+  @Test
+  public void testWagedRebalanceMetrics()
+      throws Exception {
+    _metadataStore.clearMetadataStore();
+    MetricCollector metricCollector = new WagedRebalancerMetricCollector(TEST_STRING);
+    WagedRebalancer rebalancer = new WagedRebalancer(_metadataStore, _algorithm, metricCollector);
+    // Generate the input for the rebalancer.
+    ResourceControllerDataProvider clusterData = setupClusterDataCache();
+    Map<String, Resource> resourceMap = clusterData.getIdealStates().entrySet().stream()
+        .collect(Collectors.toMap(entry -> entry.getKey(), entry -> {
+          Resource resource = new Resource(entry.getKey());
+          entry.getValue().getPartitionSet().stream()
+              .forEach(partition -> resource.addPartition(partition));
+          return resource;
+        }));
+
+    Assert.assertEquals((long) metricCollector.getMetric(
+        WagedRebalancerMetricCollector.WagedRebalancerMetricNames.GlobalBaselineCalcCounter.name(),
+        CountMetric.class).getLastEmittedMetricValue(), 0L);
+    Assert.assertEquals((long) metricCollector.getMetric(
+        WagedRebalancerMetricCollector.WagedRebalancerMetricNames.PartialRebalanceCounter.name(),
+        CountMetric.class).getLastEmittedMetricValue(), 0L);
+    // -1.0d means metric value is not set.
+    Assert.assertEquals((double) metricCollector.getMetric(
+        WagedRebalancerMetricCollector.WagedRebalancerMetricNames.BaselineDivergenceGauge.name(),
+        RatioMetric.class).getLastEmittedMetricValue(), 0.0d);
+
+    // Cluster config change will trigger baseline recalculation and partial rebalance.
+    when(clusterData.getRefreshedChangeTypes())
+        .thenReturn(Collections.singleton(HelixConstants.ChangeType.CLUSTER_CONFIG));
+
+    rebalancer.computeBestPossibleStates(clusterData, resourceMap, new CurrentStateOutput());
+
+    Assert.assertEquals((long) metricCollector.getMetric(
+        WagedRebalancerMetricCollector.WagedRebalancerMetricNames.GlobalBaselineCalcCounter.name(),
+        CountMetric.class).getLastEmittedMetricValue(), 1L);
+    Assert.assertEquals((long) metricCollector.getMetric(
+        WagedRebalancerMetricCollector.WagedRebalancerMetricNames.PartialRebalanceCounter.name(),
+        CountMetric.class).getLastEmittedMetricValue(), 1L);
+
+    // Wait for asyncReportBaselineDivergenceGauge to complete.
+    TestHelper.verify(() -> (double) metricCollector.getMetric(
+        WagedRebalancerMetricCollector.WagedRebalancerMetricNames.BaselineDivergenceGauge.name(),
+        RatioMetric.class).getLastEmittedMetricValue() == 1.0d, TestHelper.WAIT_DURATION);
```

Comment:
I believe the usage pattern is you take the return value of TestHelper.verify() and do an assertTrue. If this condition never becomes true, it would return false. Could you please take a look at the usage for this in other tests to confirm?